plugins/kubernetes-node/src/auth/PinnipedHelper.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ClusterDetails } from '@backstage/plugin-kubernetes-node';
+import * as https from 'https';
+import { bufferFromFileOrString } from '@kubernetes/client-node';
+import fetch, { RequestInit } from 'node-fetch';
+import { Logger } from 'winston';
+
+/**
+ *
+ * @public
+ */
+export type PinnipedClientCerts = {
+  key: string;
+  cert: string;
+  expirationTimestamp: string;
+};
+
+/**
+ *
+ * @public
+ */
+export type PinnipedParameters = {
+  clusterIdToken: string;
+  JWTAuthenticatorName: string;
+};
+
+type ApiResourcePinniped = {
+  authenticator: {
+    apiGroup: string;
+    kind: string;
+  };
+  apiVersion: string;
+};
```

Comment:
I'd propose merging these types to something which includes all the details necessary to make a `TokenCredentialRequest`. Also, let's use the specific term "cluster-scoped ID token" from the [Pinniped docs](https://pinniped.dev/docs/howto/configure-auth-for-webapps/#cluster-scoped-id-tokens):
```suggestion
export type PinnipedParameters = {
  clusterScopedIdToken: string;
  authenticator: {
    apiGroup: string;
    kind: string;
    name: string;
  };
  tokenCredentialRequest?: {
    apiGroup?: string;
  };
};
```